lib/graphql/batch/executor.rb

```diff
@@ -0,0 +1,38 @@
+module GraphQL::Batch
+  class Executor
+    THREAD_KEY = :"#{name}.batched_queries"
+    private_constant :THREAD_KEY
+
+    def self.current
+      Thread.current[THREAD_KEY] ||= new
+    end
+
+    attr_reader :loaders
+
+    def initialize
+      @loaders = {}
+    end
+
+    def shift
+      @loaders.shift.last
+    end
+
+    def execute(loader)
+      loader.execute
+    rescue => exc
+      loader.reject_all(exc)
```

Comment:
Yup, it does nothing if it isn't pending (https://github.com/lgierth/promise.rb/blob/v0.6.1/lib/promise.rb#L61-L75) because the [promise spec](https://promisesaplus.com/#point-14) says that a promise can't transition from the fulfilled or rejected state.